src/map/status.cpp

```diff
@@ -6036,76 +6048,87 @@ void status_calc_bl_main(struct block_list& bl, std::bitset<SCB_MAX> flag)
 		status->matk_min = status_base_matk_min(&bl, status, lv);
 		status->matk_max = status_base_matk_max(&bl, status, lv);
 
-		switch( bl.type ) {
-			case BL_PC: {
-				int wMatk = 0;
-				int variance = 0;
-
-				// Any +MATK you get from skills and cards, including cards in weapon, is added here.
-				if (sd) {
-					uint16 skill_lv;
-
-					if (sd->bonus.ematk > 0)
-						status->matk_min += sd->bonus.ematk;
-					if (pc_checkskill(sd, SU_POWEROFLAND) > 0 && pc_checkskill_summoner(sd, SUMMONER_POWER_LAND) >= 20)
-						status->matk_min += status->matk_min * 20 / 100;
-					if ((skill_lv = pc_checkskill(sd, NV_TRANSCENDENCE)) > 0)
-						status->matk_min += 15 * skill_lv + (skill_lv > 4 ? 25 : 0);
-				}
+		if (sd != nullptr) {
+			status->matk_min = status_calc_ematk(&bl, sc, status->matk_min);
+			status->matk_max = status->matk_min;
+
+			// Adds weapon magic attack (wMATK) modifications
+			// This is the only portion in MATK that varies depending on the weapon level and refinement rate.
+			if (b_status->lhw.matk) {
+				//sd->state.lr_flag = 1; //?? why was that set here
+				status->lhw.matk = b_status->lhw.matk;
+				sd->state.lr_flag = 0;
```

Comment:
Is this intended? See commented line above.

Reply:
It looks like a leftover of something, the flag is unused here